rust/src/bin/loadtest/main.rs

```diff
@@ -0,0 +1,88 @@
+mod flock;
+mod pets;
+mod stats;
+mod worker;
+
+use std::sync::Arc;
+
+use log::*;
+use scylla::Session;
+use structopt::StructOpt;
+
+use care_pet::{db, Result};
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "loadtest")]
+struct App {
+    // number of the startPets owners
+    #[structopt(long, default_value = "100")]
+    owners: usize,
+
+    // number of startPets to simulate
+    #[structopt(long, default_value = "100")]
+    pets: usize,
+
+    // number of sensors per pet
+    #[structopt(long, default_value = "4")]
+    sensors: usize,
+
+    // an interval between sensors measurements
+    #[structopt(long, default_value = "1s")]
+    interval: humantime::Duration,
+
+    // just write random data
+    #[structopt(long)]
+    writer: bool,
+
+    // number of parallel writers: Default: num of CPUs
+    #[structopt(long)]
+    workers: Option<usize>,
+
+    #[structopt(flatten)]
+    db_config: db::Config,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    care_pet::log::init();
+
+    let app: App = App::from_args();
+    debug!("Configuration = {:?}", app);
+
+    info!("Welcome to the Pets simulator");
+    rlimit::utils::increase_nofile_limit(102400).unwrap();
+
+    // an easy way to get a static lifetime Session without wrapping it in an Arc
+    // this is fine since we know that the session will last for the lifetime
+    // of the program and that the Session is thread safe
+    let sess: &'static Session = Box::leak(Box::new(
+        db::new_session_with_keyspace(&app.db_config).await?,
+    ));
+
+    info!("Creating flock");
+    let f = flock::new(app.owners, app.pets, app.sensors);
+
+    flock::save(sess, &f).await?;
+
+    let workers = app.workers.unwrap_or_else(num_cpus::get);
+    if app.writer {
+        let sensors = Arc::new(f.2);
+        for i in 1..=workers {
+            let sensors = sensors.clone();
+            tokio::spawn(async move {
+                worker::work(sess, i, sensors).await;
+            });
+        }
+
+        info!("Writers started");
+    } else {
+        pets::start(sess, app.interval, f);
+        info!("Flock started");
+    }
+
+    // the application can only be terminated with interrupt signals, nothing to wait for
+    // specifically, so waiting for a channel that will never receive a value is fine
+    let (tx, rx) = tokio::sync::oneshot::channel::<()>();
+    std::mem::forget(tx);
+    rx.await.map_err(From::from)
```

Comment:
That snippet looks quite complicated. I think it's an equivalent of
```rust
let () = std::future::pending().await;
```
, which is quite self-descriptive and doesn't need the mem::forget trick.